### ISSUE TITLE
Common errors when upgrading to 2.0.12050 (or newer)

### ISSUE THREAD

**Issue by paulbatum** [Azure/azure-functions-host#3363]
This relates to this announcement: https://github.com/Azure/app-service-announcements/issues/129

### Check your versions
First, to understand any error you're getting, its important that you establish which version(s) of functions you are running. If you're working completely from within the portal its simple - go to function app settings and look at the runtime version:

![image](https://user-images.githubusercontent.com/10663/44889884-ced0f600-ac8c-11e8-87bf-f427f1d23fd1.png)

If you're developing locally, you'll also need to check the functions version that is being used when you test your app locally. You can check that by looking for a line in the output that looks like this:
[8/30/2018 6:12:53 PM] Starting Host (HostId=xxx, InstanceId=xxx, Version=2.0.12050.0)

If your local version doesn't match the version you're deploying to in Azure, you're probably going to have issues.

### Check your logs
If you're developing and testing the app locally, the logs are clearly visible in the terminal and easy to check. If you've deployed to Azure and you're having problems, there are a few approaches you can take to get quick feedback on what is going wrong with your function app:
1. Go to a particular function in the portal and attempt to run it and check the log stream at the bottom of the screen.
2. Go to the overview tab, and in configured features select Application Insights. This will open a new view. Find Live Metrics Stream in the left hand list, open that and look at the Sample Telemetry - this will show live error logs for the entire function app (not just a single function)
3. Go to the platform features tab and select Advanced Tools (kudu). Go to debug console -> CMD and then in the file browser browse to Log Files -> Application -> Functions -> Host and look at the log files there.

#### [Error] Found functions with more than one language. Select a language for your function app by specifying FUNCTIONS_WORKER_RUNTIME AppSetting

You'll need to specify the language for your function app. See the `Choose a language for your Function App` section of [this page](https://github.com/Azure/app-service-announcements/issues/129).

#### My http triggers are returning 404 but I have no idea why
Its possible this is happening because you have the same error as the one above (multiple languages), but you're just not seeing it in the log. Try step approach 3 in the "check your logs" section above to confirm.

#### Microsoft.Azure.WebJobs.Script.HostConfigurationException : The host.json file is missing the required 'version' property.

You'll need to update your host.json file to the new format. See [here](https://aka.ms/functions-hostjson) for more info.

#### System.TypeLoadException : Could not load type 'Microsoft.Azure.WebJobs.ExecutionContext' from assembly 'Microsoft.Azure.WebJobs.Extensions, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null'.

You need to update your NuGet references, recompile and publish. See the `Update your dependencies, extensions and/or code` section of [this page](https://github.com/Azure/app-service-announcements/issues/129).

#### The binding type(s) 'blob, queue' are not registered. Please ensure the type is correct and the binding extension is installed.

You'll get errors like these if you're using Azure Storage based functionality (blob/queue/tables trigger/input/output) and you haven't installed the `Microsoft.Azure.WebJobs.Extensions.Storage` extension. See the `Update your dependencies, extensions and/or code` section of [this page](https://github.com/Azure/app-service-announcements/issues/129).

#### Could not load type 'Microsoft.Azure.WebJobs.Hosting.IWebJobsStartup' from assembly 'Microsoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null'.

You'll see this if you have published an app that has been updated locally to the latest bits, but the app you're deploying to is running a version older than `2.0.12050.0`. See the instructions at the top of this page for checking your versions.

#### Microsoft.Azure.WebJobs.Script.Rpc.LanguageWorkerChannelException : Failed to start language worker for: node.  Error: Your Function App is currently set to use Node.js version v6.5.0, but the runtime requires an Active LTS or Current version (ex: 8.11.1 or 10.6.0).

The JavaScript support in functions V2 works on Node 8 and Node 10. See [here](https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-node#node-version-and-package-management) for more info.

**Comment by kgoderis** [comment]
@paulbatum After upgrading, and within the VS Code IDE, all TraceWriter.Info() calls are written twice to the Terminal that is running func. Once in blue, with inclusion of a date/time stamp, as you would expect (e.g. as before upgrading), once in white, as if it is Console.WriteLine()

**Comment by kgoderis** [comment]
@paulbatum See https://github.com/Azure/homebrew-functions/issues/10 . I had to download the .zip manually, and switch over symlinks in order to make it work.

**Comment by harjotsmart** [comment]
After updating to new version on local, update all the nugets, when run getting this one.
I m using custom IExtensionConfigProvider to inject interaces on Intialize. 
 
[31-08-2018 14:13:26] Function 'ApiAppService.RunGetAppServices' failed indexing and will be disabled.
[31-08-2018 14:13:26] Error indexing method 'ApiAppService.RunAppServiceDomain'
[31-08-2018 14:13:26] Microsoft.Azure.WebJobs.Host: Error indexing method 'ApiAppService.RunAppServiceDomain'. Microsoft.Azure.WebJobs.Host: Cannot bind parameter 'faaslogger' to type ILogger. Make sure the parameter Type is supported by the binding. If you're using binding extensions (e.g. ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. config.UseServiceBus(), config.UseTimers(), etc.).

**Comment by davemurphysf** [comment]
I was able to upgrade successfully but I started seeing these dependency errors for the logs.  I tried turning log storage on/off and changing containers with no luck.  My functions are working fine, but thought I'd mention this.
<img width="530" alt="screen shot 2018-08-31 at 11 54 48 am" src="https://user-images.githubusercontent.com/408705/44930919-be645e00-ad14-11e8-9bd1-5e52f39d74a7.png">

**Comment by DanRigby** [comment]
I'm seeing the same dependency errors as @davemurphysf along with a few other dependency errors with 409s when attempting to write to `https://<function-name>.blob.core.windows.net:443/azure-webjobs-hosts/locks/linkfunctions/host?comp=lease`

Both sets of errors are intermittent.

**Comment by harjotsmart** [comment]
@paulbatum can u please guide, after migration 
 public void Initialize(ExtensionConfigContext context) 
is not called which was working earlier. because of this, dependencies are not loading on startup.

**Comment by hoovercj** [comment]
I attempted to follow the migration steps but still get `azure functions binding type "blob" not registered` in my node/javascript functions app.

In response to that issue, the OP says:

> You'll get errors like these if you're using Azure Storage based functionality (blob/queue/tables trigger/input/output) and you haven't installed the Microsoft.Azure.WebJobs.Extensions.Storage extension. See the Update your dependencies, extensions and/or code section of this [page](https://github.com/Azure/app-service-announcements/issues/129).

Following those instructions, in the _root_ directory of my functions app, I ran: `func extensions install --package Microsoft.Azure.WebJobs.Extensions.Storage --version 3.0.0-beta8` which simply added a file called `extensions.csproj` in the root of my project with these contents:

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
	<WarningsAsErrors></WarningsAsErrors>
	<DefaultItemExcludes>**</DefaultItemExcludes>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Storage" Version="3.0.0-beta8" />
    <PackageReference Include="Microsoft.Azure.WebJobs.Script.ExtensionsMetadataGenerator" Version="1.0.0-beta3" />
  </ItemGroup>
</Project>
```

I added that file and published from github, but I still get the error.

Unfortunately all the documentation, including [Azure Table storage bindings for Azure Functions](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-table#packages---functions-2x), is out of date and does not yet reflect the need to register bindings so it is unclear if there are other necessary steps.

I have have left the code in the state that I tried to use with the newest runtime, but I have pinned the function version in the application settings as a mitigation for now.

**Comment by hoovercj** [comment]
Somebody created a [PR for my repo](https://github.com/hoovercj/strava-weather-azure-functions/pull/5) setting the version value in `host.json` and correcting the cli-generated version of a package in the `.csproj` file which solved my initial issue (`azure functions binding type "blob" not registered`).

However, this quote from the [migration notes](https://github.com/Azure/app-service-announcements/issues/129) is a bit confusing about whether updating the hosts file is necessary or not:

> The "version": "2.0" setting is now required in host.json. If there are no current settings in the file, the host will automatically add this value.

Based on the quote above, it sounds like the version shouldn't be necessary. So maybe the issue was just the package version in the `.csproj` file?

But now I have 2 more issues:
1) After modifying my `.csproj` and  `host.json` files, all my functions timeout. There were no source changes to the individual functions between upgrading, only touching the `.csproj` and `host.json`. Did the runtime change how it handles returned promises in javascript functions? Do I need to explicitly call resolve instead of using async/returning promises?

2) If I roll back the changes to the `csproj` and `host.json` files and pin the version, I get `Could not load type 'Microsoft.Azure.WebJobs.Hosting.IWebJobsStartup' from assembly 'Microsoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null'.`. The post above says:

>You'll see this if you have published an app that has been updated locally to the latest bits, but the app you're deploying to is running a version older than 2.0.12050.0

However, I undid the changes and then even redeployed the last known working version from the azure portal "Deployment Options" blade and I **still** get this issue.

### Summary
I have 3 issues:
1) Documentation is unclear about whether `host.json` needs modified / what the version setting will be if it is omitted
2) Javascript functions that worked in v2 before the update now timeout after updating
3) Reverting back, even to the last known good deployment with a pinned version, results in  `Could not load type 'Microsoft.Azure.WebJobs.Hosting.IWebJobsStartup' from assembly 'Microsoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null'.`
![image](https://user-images.githubusercontent.com/2957173/44971469-f1496480-af55-11e8-9a49-4136a1181d93.png)

**Comment by harshithkashyap** [comment]
@paulbatum We are also seeing the same HTTP 409 dependency errors as mentioned by @DanRigby  [above](https://github.com/Azure/azure-functions-host/issues/3363#issuecomment-417763532) across all our function apps. We've temporarily set functions runtime to `2.0.11961-alpha` but would request a solution for this issue. All our functions are coded in JavaScript and deployed using Zip Push Deploy + Run from Zip if that helps.

**Comment by kgoderis** [comment]
https://github.com/Azure/azure-functions-host/issues/3367 is still prevailing after upgrading to 2.0.1-beta.37.

**Comment by jonbjo** [comment]
Azure Functions Core Tools (2.0.1-beta.37)
Function Runtime Version: 2.0.12050.0

I have an issue after the upgrade which I've managed to boil down to something that is easy to reproduce. A function with ServiceBusTrigger that runs longer than 5 minutes **succeeds** but fails directly after that when trying to complete the trigger message.

```
[09/03/2018 19:57:03] Executing 'ErrorActionComplete' (Reason='New ServiceBus message detected on 'myTopic/Subscriptions/mySubscription'.', Id=54417f8c-e285-4c52-8057-606d647b5968)
[09/03/2018 19:57:03] Time elapsed: 0 minutes
Time elapsed: 0 minutes
[09/03/2018 19:58:03] Time elapsed: 1 minutes
Time elapsed: 1 minutes
[09/03/2018 19:59:03] Time elapsed: 2 minutes
Time elapsed: 2 minutes
[09/03/2018 20:00:03] Time elapsed: 3 minutes
Time elapsed: 3 minutes
[09/03/2018 20:01:03] Time elapsed: 4 minutes
Time elapsed: 4 minutes
[09/03/2018 20:02:03] Time elapsed: 5 minutes
Time elapsed: 5 minutes
[09/03/2018 20:03:03] Executed 'ErrorActionComplete' (Succeeded, Id=54417f8c-e285-4c52-8057-606d647b5968)
[09/03/2018 20:03:04] MessageReceiver error (Action=Complete, ClientId=MessageReceiver1myTopic/Subscriptions/mySubscription, EntityPath=myTopic/Subscriptions/mySubscription, Endpoint=test-prod.servicebus.windows.net)
[09/03/2018 20:03:04] Microsoft.Azure.ServiceBus: The lock supplied is invalid. Either the lock expired, or the message has already been removed from the queue. Reference:b108c442-3fc2-4108-9718-a625f01b795b, TrackingId:2869602600ab7d4100000f035b8d9209_G1_B15, SystemTracker:test-prod:Topic:mytopic|mysubscription, Timestamp:9/3/2018 8:03:03 PM.

```
Below function works like expected when loop is less than 5 minutes but not when running longer.

```dotnet
[FunctionName("ErrorActionComplete")]
public static async Task Run(
    [ServiceBusTrigger(topicName: "myTopic", subscriptionName: "mySubscription", Connection = "ServiceBusConnection")]string message,
    TraceWriter logger)
{
    for (int i = 0; i < 6; i++)
    {
        logger.Info("Time elapsed: " + i + " minutes");
        Thread.Sleep(60000);
    }
}
```
host.json
```javascript
{
  "version": "2.0",
  "extensions": {
    "serviceBus": {
      "messageHandlerOptions": {
        "maxConcurrentCalls": 1,
        "maxAutoRenewTimeout": "00:30:00"
      }
    }
  }
}
```

Before the latest release I had the issue that the lock wasn't auto-renewed after 5 minute and hence started a new execution in parallel despite having maxConcurrency set to 1, similar to
https://github.com/Azure/azure-functions-host/issues/2667. Now it looks like that is fixed and the lock is auto-renewed, but message can't be completed when finished.

**Comment by cristinamanitiu** [comment]
Hello,

I have an HTTP triggered function which worked until the newest runtime update. Now it sometimes runs and sometimes it responds with 500 Internal Server Error. 

The Azure runtime on which it runs is: 2.0.12050.0 (beta).
I have added the FUNCTIONS_WORKER_RUNTIME app setting with value "java".
On the logs I see the following error:

`Microsoft.Azure.WebJobs.Host.FunctionInvocationException:
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<ExecuteWithLoggingAsync>d__17.MoveNext (Microsoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=nullMicrosoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null: C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.csMicrosoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null: 293)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<TryExecuteAsync>d__14.MoveNext (Microsoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=nullMicrosoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null: C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.csMicrosoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null: 89)
Inner exception Microsoft.Azure.WebJobs.Script.Rpc.LanguageWorkerChannelException handled at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw:
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Script.Description.WorkerLanguageInvoker+<InvokeCore>d__6.MoveNext (Microsoft.Azure.WebJobs.Script, Version=2.0.0.0, Culture=neutral, PublicKeyToken=nullMicrosoft.Azure.WebJobs.Script, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null: C:\azure-webjobs-sdk-script\src\WebJobs.Script\Description\Rpc\WorkerLanguageInvoker.csMicrosoft.Azure.WebJobs.Script, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null: 74)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Script.Description.FunctionInvokerBase+<Invoke>d__24.MoveNext (Microsoft.Azure.WebJobs.Script, Version=2.0.0.0, Culture=neutral, PublicKeyToken=nullMicrosoft.Azure.WebJobs.Script, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null: C:\azure-webjobs-sdk-script\src\WebJobs.Script\Description\FunctionInvokerBase.csMicrosoft.Azure.WebJobs.Script, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null: 84)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Script.Description.FunctionGenerator+<Coerce>d__3`1.MoveNext (Microsoft.Azure.WebJobs.Script, Version=2.0.0.0, Culture=neutral, PublicKeyToken=nullMicrosoft.Azure.WebJobs.Script, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null: C:\azure-webjobs-sdk-script\src\WebJobs.Script\Description\FunctionGenerator.csMicrosoft.Azure.WebJobs.Script, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null: 225)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionInvoker`2+<InvokeAsync>d__9.MoveNext (Microsoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=nullMicrosoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null: C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionInvoker.csMicrosoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null: 63)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<InvokeAsync>d__25.MoveNext (Microsoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=nullMicrosoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null: C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.csMicrosoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null: 561)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<ExecuteWithWatchersAsync>d__24.MoveNext (Microsoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=nullMicrosoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null: C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.csMicrosoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null: 508)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<ExecuteWithLoggingAsync>d__23.MoveNext (Microsoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=nullMicrosoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null: C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.csMicrosoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null: 444)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult (System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor+<ExecuteWithLoggingAsync>d__17.MoveNext (Microsoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=nullMicrosoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null: C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.csMicrosoft.Azure.WebJobs.Host, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null: 249)
Inner exception System.AggregateException handled at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw:
Microsoft.Azure.WebJobs.Script.Rpc.LanguageWorkerProcessExitException:
System.TimeoutException:

> Microsoft.Azure.WebJobs.Script.Rpc.LanguageWorkerProcessExitException:`

**Comment by brettsam** [comment]
@davemurphysf -- Those dependency errors look to be coming from the Azure Functions dashboard. Do you have the `AzureWebJobsDashboard` app setting set? If you remove it, do those go away? Ideally those wouldn't show up, even if you had this enabled -- I've filed https://github.com/Azure/azure-webjobs-sdk/issues/1876 to track this.

@DanRigby -- What trigger are you using with the function that is showing your lease dependency error? Or are you using a `[Singleton]` attribute directly? It's similar to the issue above but I'll likely want to track it separately as a different piece of code is adding that.

**Comment by brettsam** [comment]
@harjotsmart -- See this comment for changes required to existing binding extensions: https://github.com/Azure/app-service-announcements-discussions/issues/64#issuecomment-417723700

**Comment by brettsam** [comment]
@hoovercj -- The host.json change is not strictly required. An empty host.json is fine. We want to be clear that if you're using custom settings, you need to explicitly acknowledge that there is a change in v2. Since you're not using any custom settings, you're fine. 

@mhoeger,  can you help with the question about promises/done/etc? https://github.com/Azure/azure-functions-host/issues/3363#issuecomment-418016714

**Comment by IGx89** [comment]
@harshithkashyap We're also using Zip Push Deploy + Run from Zip and have been broken since this update. The "Functions (Read Only)" node in the Portal just hangs and never expands the functions. Anyone else using a similar deployment method have any luck getting it to work?

**Comment by kgoderis** [comment]
@IGx89 What worked for me was first stopping the Function in the Portal, then deploying, and then restarting the Function in the portal

**Comment by IGx89** [comment]
@kgoderis Good suggestion, but no luck -- same error message. It seems to be running based on all the 404/409 Azure Storage requests in AI (which @brettsam says is normal), but not being able to see individual functions in the Portal or manually trigger them is a pretty big problem.

**Comment by brettsam** [comment]
@IGx89 -- can you share your site name with me so I can investigate on our side? See here to share privately: https://github.com/Azure/azure-functions-host/wiki/Sharing-Your-Function-App-name-privately. Although if you're not even able to run a function, maybe that's tough. Any older logs you have would work as well -- I should be able to look up any function invocation guid.

If you need to, you can email me your site name directly. My mail address is in my github profile.

**Comment by ltouro** [comment]
Can't create a blank function app with ~2 version pinned and a basic HTTP trigger. Everything through the interface. The message is: Function (HttpTriggerJS1) Error: Failed to start language worker process for: node. node exited with code 1 .

Also, trying to upgrade an existing app blew it. Returning to the previous pinned version has no effect.

**Comment by mhoeger** [comment]
@ltouro - what version of node are you using? (note that if your are specifying node version in azure using WEBSITE_NODE_DEFAULT_VERSION, only a few versions are supported. i would recommend trying 8.11.1 or 10.6.0).

**Comment by pragnagopa** [comment]
@cristinamanitiu - We are tracking issue with Java process timeout exception on consumption plan. Please see https://github.com/Azure/azure-functions-host/issues/3081 I will continue to follow up on this issue.

**Comment by nbevans** [comment]
```
Azure Functions Core Tools (2.0.1-beta.35)
Function Runtime Version: 2.0.11960.0
```

How do I get this Function Runtime Version to update?

**Comment by nbevans** [comment]
Seems chocolately hasn't updated to beta.37 yet and that was why. Had to install node/npm and get it that way. Remind me why Azure Functions has an "official dependency" on the god-awful Node/NPM stack again? It is beyond bizarre!

**Comment by makar-sasha** [comment]
Hello. 
After an update to the recent functions version App Insights is not storing the data. I see the logs in Live Metrics Stream and looks like the functions working correctly. But all other App Insights views are empty so the logs were not delivered. Maybe there are some advise how to fix the problem?

**Comment by ltouro** [comment]
@makar-sasha I was about to report the same issue. Can see the logs on live stream but no query results on the analytics dashboard. Runtime: Node v. 8.11.1

**Comment by davidebbo** [comment]
There are too many distinct things being discussed in this thread, and it's hard to track/respond properly. Please open a separate issue on https://github.com/Azure/azure-functions-host for each distinct issue that you are having (you can post a link from here). We will get to all of them. Thanks!

**Comment by hoovercj** [comment]
@brettsam thanks, I have confirmed that simply adding the `extensions.csproj` file without modifying `host.json` was sufficient.

I have opened #3391 to address my async issues.

**Comment by davidebbo** [comment]
@nbevans Chocolatey has been updated with the latest.

**Comment by betarabbit** [comment]
I have upgraded a java version of function app to `Runtime version: 2.0.12050.0 (beta)` and followed [instructions](https://github.com/Azure/azure-functions-host/wiki/Updating-your-App-(portal)#update-your-extensions) to add extensions for storage queueTrigger, but, after restarted function app, it seems all request to httpTrigger not working any more.

**Comment by davidebbo** [comment]
@betarabbit please open a separate issue, per my [previous comment](https://github.com/Azure/azure-functions-host/issues/3363#issuecomment-418801752).

**Comment by mjul** [comment]
Related issue #3411 listSecrets broken for function keys in V2 (Runtime version: 2.0.12050.0)

**Comment by TurtleBeach** [comment]
I am both puzzled and frustrated with inconsistent and (apparently) meaningless errors now being generated by maven and VS Code builds.
I have updated to the 2.0.2 core function tools, yet maven and VS Code builds complain and imply that the core tools were not found:

[INFO] Step 7 of 7: Installing function extensions if needed
[WARNING] Failed to get Azure Functions Core Tools version locally
[WARNING] Local Azure Functions Core Tools does not support extension auto-install, skip it in the package phase.
[WARNING] Local version of Azure Functions Core Tools (null) does not match the latest (2.0.2). Please update it for the best experience. See: https://aka.ms/azfunc-install
[INFO] Function extension installation done.
[INFO] Successfully built Azure Functions.

local.settings.json looks like this (after supposedly installing java extension)
{
  "IsEncrypted": false,
  "Values": {
	"AzureWebJobsStorage": "",
    "AzureWebJobsDashboard": "",
	"FUNCTION_WORKER_RUNTIME": "java"
  },
  "ConnectionStrings": {}
}

BUT when I run the package (locally so far) it reports:

[INFO] Azure Function App's staging directory found at: D:\Functions\neuralstudio-create-container\target\azure-functions\neuralstudio-create-container
[INFO] Azure Functions Core Tools found.
your worker runtime is not set. As of 2.0.1-beta.26 a worker runtime setting is required.
Please run `func settings add FUNCTIONS_WORKER_RUNTIME <option>` or add FUNCTIONS_WORKER_RUNTIME to your local.settings.json
Available options: dotnet, node

< graphic omitted >

Azure Functions Core Tools (2.0.2.0)
Function Runtime Version: 2.0.12115.0

and a few lines later:
[9/20/2018 5:51:50 PM] Initializing Host.
[9/20/2018 5:51:50 PM] Host initialization: ConsecutiveErrors=0, StartupCount=1
[9/20/2018 5:51:50 PM] Starting JobHost
[9/20/2018 5:51:50 PM] Starting Host (HostId=neuralstudio-869013314, InstanceId=631b09b3-8e17-48a2-9c7e-7cf6d0d5a93d, Version=2.0.12115.0, ProcessId=10768, AppDomainId=1, Debug=False, FunctionsExtensionVersion=)

Then the function runs as expected.

Development on Azure is difficult enough without this kind of aggravation - I spent a couple of hours trying to figure out what was going on before I realized the function was working. But now I fear that bad things will happen if I in fact deploy - I am particularly worried that there should be a value for "FunctionsExtensionVersion" in the last line shown above.

What is going on?
Jack

**Comment by TurtleBeach** [comment]
In retrospect my comment above should have been a new issue not a comment on list secrets broken.
Apologies.

**Comment by pragnagopa** [comment]
@TurtleBeach - We found  issue with core tools version.  Will roll out a fix.

**Comment by pragathikadaganchi** [comment]
I have same error 
Microsoft.Azure.WebJobs.Script.Rpc.LanguageWorkerProcessExitException.


Event time | 9/25/2018, 4:36:28 PM |  
-- | -- | --
Message | D:\Program Files\FunctionsJava\zulu8.30.0.1-jdk8.0.172-win_x64\bin\java exited with code -1 |  
Exception type | Microsoft.Azure.WebJobs.Script.Rpc.LanguageWorkerProcessExitException |  
Failed method | Unknown |  
Sample rate | 1 |  
Telemetry type | exception
Telemetry type | exception |  
-- | -- | --
Device type | PC |  
Client IP address | 0.0.0.0 |  
SDK version | azurefunctions: 2.0.12115.0 |  
Cloud role name | pragathistagging |  
Cloud role instance | b6d9dfe51aa28a85d17ea7ce379631fadc8ba4df84486d78a980d26ff66b9ad6 |  
Severity level | Error |  
Problem Id | Microsoft.Azure.WebJobs.Script.Rpc.LanguageWorkerProcessExitException |  
Assembly | Unknown

Language used is java.

**Comment by ZombiesWithCoffee** [comment]
I'm receiving "The type or namespace name 'JobHostConfiguration' could not be found."

Which assembly did that method move to?

**Comment by ZombiesWithCoffee** [comment]
This code no longer compiles.  
Where is JobHostConfiguration? I searched and couldn't find it on nuget
Where did RunAndBlock go?  All I see now is StartAsync.
The method is now async.   How should I change the call?

            var config = new JobHostConfiguration {
                JobActivator = new SimpleInjectorActivator(container),
                NameResolver = new QueueNameResolver()
            };

            if (config.IsDevelopment) {
                config.UseDevelopmentSettings();
            }

            var host = new JobHost(config);
            host.RunAndBlock();

**Comment by brettsam** [comment]
@ZombiesWithCoffee  -- WebJobs v3 now uses .NET Core hosting patterns with `HostBuilder` to construct and start the host. See our WebJobs sample here for an example: https://github.com/Azure/azure-webjobs-sdk/blob/dev/sample/SampleHost/Program.cs

**Comment by ZombiesWithCoffee** [comment]
Thank you @brettsam!  That's a major update.

**Comment by scott-lin** [comment]
Let me know if I should open a separate issue, but this seemed like a good place.

I'm migrating from function runtime version `2.0.11961-alpha` to `2.0.12134.0 (~2)` now that GA for 2.0 has released. My function uses a queue trigger, so I've installed the `Microsoft.Azure.WebJobs.Extensions.Storage` extension via the Portal prompts. I've waited over 10 minutes for it to complete.

Now I'm seeing the following error toast in the Portal:
```
The function runtime is unable to start. System.Private.CoreLib: Exception has been thrown by the target of an invocation. Microsoft.Azure.WebJobs.Extensions.Storage: Specified argument was out of the range of valid values.
 Parameter name: value.
```

I believe I've followed all of the migration instructions, so I'm at a loss right now. Is something wrong with my Function settings, or do I need to take a code change?

**Function Information**
- Function App name: devautorest-functions
- Function name: autorestprocessor
- Region: west us 2

**host.json File**
```
{
  "version": "2.0",
  "functionTimeout": "00:10:00",
  "extensions": {
        "queues": {
            "maxDequeueCount": 5,
            "batchSize": 1,
            "newBatchThreshold": 0
        }
    }
}
```

**Queue Binding**
```
{
    "bindings": [
        {
            "name": "clientLibraryId",
            "type": "queueTrigger",
            "direction": "in",
            "queueName": "clientlibraryrequest",
            "connection": "STORAGEACCOUNT_CONNECTIONSTRING"
        }
    ],
    "disabled": false
}
```

**Comment by brettsam** [comment]
I'm going to go ahead and close this as we're now sufficiently past this release. If you find any new problems, please create a new issue and we will triage.
